packages/loot-core/src/platform/client/fetch/index.d.ts

```diff
@@ -7,25 +7,31 @@ export type Init = typeof init;
 
 export function send<K extends keyof Handlers>(
   name: K,
-  args?: Parameters<Handlers[K]>[0],
-  options?: { catchErrors: true },
-): ReturnType<
-  | { data: Handlers[K] }
-  | { error: { type: 'APIError' | 'InternalError'; message: string } }
+  args: Parameters<Handlers[K]>[0],
+  options: { catchErrors: true },
+): Promise<
+  | { data: Awaited<ReturnType<Handlers[K]>>; error: undefined }
+  | {
+      data: undefined;
+      error: { type: 'APIError' | 'InternalError'; message: string };
+    }
 >;
 export function send<K extends keyof Handlers>(
   name: K,
   args?: Parameters<Handlers[K]>[0],
   options?: { catchErrors?: boolean },
-): ReturnType<Handlers[K]>;
+): Promise<Awaited<ReturnType<Handlers[K]>>>;
 export type Send = typeof send;
 
 export function sendCatch<K extends keyof Handlers>(
   name: K,
   args?: Parameters<Handlers[K]>[0],
-): ReturnType<
-  | { data: Handlers[K] }
-  | { error: { type: 'APIError' | 'InternalError'; message: string } }
+): Promise<
+  | { data: Awaited<ReturnType<Handlers[K]>>; error: undefined }
```

Comment:
```suggestion
  | { data: Awaited<ReturnType<Handlers[K]>> }
```

Reply:
If I remove this I get `Property 'error' does not exist on type` in a bunch of places like this:
```tsx
const { data, error } = await sendCatch('gocardless-get-banks', country);
```
I don't see any other way to keep the ability to destructure like this, any ideas?

Reply:
I see. In that case we should probably just merge those two properties and make them both optional. WDYT?

Reply:
Hmm, I kind of like being able to assert that `data` is `undefined` if `error` is not, and vice-versa. I think it would make the code a bit messy otherwise unfortunately as you'd have to check both